ocb3/src/lib.rs

```diff
@@ -87,17 +124,23 @@ type Sum = GenericArray<u8, SumSize>;
 impl<Cipher, NonceSize, TagSize> KeySizeUser for Ocb3<Cipher, NonceSize, TagSize>
 where
     Cipher: KeySizeUser,
-    TagSize: ArrayLength<u8> + IsGreater<U0> + IsLessOrEqual<U16>,
     NonceSize: ArrayLength<u8> + IsGreaterOrEqual<U6> + IsLessOrEqual<U15>,
+    TagSize: ArrayLength<u8> + NonZero + IsLessOrEqual<U16>,
+    GrEq<NonceSize, U6>: NonZero,
+    LeEq<NonceSize, U15>: NonZero,
+    LeEq<TagSize, U16>: NonZero,
```

Comment:
Note that these bounds cause quite ugly compilation errors:
```
error[E0599]: the function or associated item `new` exists for struct `Ocb3<Aes128, UInt<UInt<..., ...>, ...>, ...>`, but its trait bounds were not satisfied
 --> ocb3/src/lib.rs:100:40
  |
9 | let cipher = Ocb3::<Aes128, U12, U20>::new(&key);
  |                                        ^^^ function or associated item cannot be called on `Ocb3<Aes128, UInt<UInt<..., ...>, ...>, ...>` due to unsatisfied trait bounds
  |
  = note: the full type name has been written to '/tmp/rustdoctestWBN6Uv/rust_out.long-type-4996721529537771483.txt'
  = note: consider using `--verbose` to print the full type name to the console
  = note: the following trait bounds were not satisfied:
          `&Ocb3<Aes128, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>, UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B1>, B0>, B0>>: InnerInit`
          which is required by `&Ocb3<Aes128, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>, UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B1>, B0>, B0>>: KeyInit`
          `&Ocb3<Aes128, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>, UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B1>, B0>, B0>>: InnerUser`
          which is required by `&Ocb3<Aes128, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>, UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B1>, B0>, B0>>: KeyInit`
          `&mut Ocb3<Aes128, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>, UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B1>, B0>, B0>>: InnerInit`
          which is required by `&mut Ocb3<Aes128, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>, UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B1>, B0>, B0>>: KeyInit`
          `&mut Ocb3<Aes128, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>, UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B1>, B0>, B0>>: InnerUser`
          which is required by `&mut Ocb3<Aes128, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>, UInt<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B1>, B0>, B0>>: KeyInit`

error[E0277]: the trait bound `B0: aes::cipher::typenum::NonZero` is not satisfied
   --> ocb3/src/lib.rs:100:14
    |
9   | let cipher = Ocb3::<Aes128, U12, U20>::new(&key);
    |              ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `aes::cipher::typenum::NonZero` is not implemented for `B0`
    |
    = help: the following other types implement trait `aes::cipher::typenum::NonZero`:
              B1
              PInt<U>
              NInt<U>
              UInt<U, B>
note: required by a bound in `Ocb3`
   --> /home/newpavlov/projects/rust/crypto/AEADs/ocb3/src/lib.rs:108:25
    |
102 | pub struct Ocb3<Cipher, NonceSize = U12, TagSize = U16>
    |            ---- required by a bound in this struct
...
108 |     LeEq<TagSize, U16>: NonZero,
    |                         ^^^^^^^ required by this bound in `Ocb3`

```

Reply:
Ok, that might be a good reason to use sealed traits for now